mungegithub/mungers/approvers/owners.go

```diff
@@ -0,0 +1,200 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package approvers
+
+import (
+	"strings"
+	"sort"
+	"math/rand"
+
+	"k8s.io/kubernetes/pkg/util/sets"
+	"k8s.io/test-infra/mungegithub/features"
+
+)
+
+const (
+	ownersFileName = "OWNERS"
+)
+
+
+type RepoInterface interface {
+	Approvers(path string) sets.String
+	LeafApprovers(path string) sets.String
+	FindOwnersForPath(path string) string
+}
+
+type RepoAlias struct {
+	repo  RepoInterface
+	alias features.Aliases
+}
+
+func NewRepoAlias(repo RepoInterface, alias features.Aliases) RepoAlias {
+	return RepoAlias{
+		repo:  repo,
+		alias: alias,
+	}
+}
+
+func (r RepoAlias) Approvers(path string) sets.String {
+	return r.alias.Expand(r.repo.Approvers(path))
+}
+
+func (r RepoAlias) LeafApprovers(path string) sets.String {
+	return r.alias.Expand(r.repo.LeafApprovers(path))
+}
+
+func (r RepoAlias) FindOwnersForPath(path string) string {
+	return r.repo.FindOwnersForPath(path)
+}
+
+type Owners struct {
+	filenames []string
+	repo RepoInterface
+	seed int64
+}
+
+func (o Owners) NewOwners(filenames[]string, r RepoInterface, s int64) Owners {
+	return Owners{filenames: filenames, repo: r, seed: s}
+}
+
+
+// removeSubdirs takes a list of directories as an input and returns a set of directories with all
+// subdirectories removed.  E.g. [/a,/a/b/c,/d/e,/d/e/f] -> [/a, /d/e]
+func removeSubdirs(dirList []string) sets.String {
+	toDel := sets.String{}
+	for i := 0; i < len(dirList)-1; i++ {
+		for j := i + 1; j < len(dirList); j++ {
+			// ex /a/b has prefix /a so if remove /a/b since its already covered
+			if strings.HasPrefix(dirList[i], dirList[j]) {
+				toDel.Insert(dirList[i])
+			} else if strings.HasPrefix(dirList[j], dirList[i]) {
+				toDel.Insert(dirList[j])
+			}
+		}
+	}
+	finalSet := sets.NewString(dirList...)
+	finalSet.Delete(toDel.List()...)
+	return finalSet
+}
+
+func (o Owners) GetApprovers() map[string]sets.String {
+	ownersToApprovers := map[string]sets.String{}
+
+	for _, fn := range o.ListOwnersFiles() {
+		ownersToApprovers[fn] = o.repo.LeafApprovers(fn)
+	}
+
+	return ownersToApprovers
+}
+
+func (o Owners) GetReverseMap() map[string]sets.String {
+	approverOwnersfiles := map[string]sets.String{}
+	for ownersFile, approvers := range o.GetApprovers() {
+		for approver := range approvers {
+			if _, ok := approverOwnersfiles[approver]; ok {
+				approverOwnersfiles[approver].Insert(ownersFile)
+			} else {
+				approverOwnersfiles[approver] = sets.NewString(ownersFile)
+			}
+		}
+	}
+	return approverOwnersfiles
+}
+
+func (o Owners) GetShuffledApprovers() (people []string, files []sets.String) {
```

Comment:
Let's just return the list of people. That list and the reverse map is good enough I think